sources/Renderer/OpenGL/Platform/Linux/LinuxGLContext.cpp

```diff
@@ -205,57 +280,217 @@ bool LinuxGLContext::SetSwapInterval(int interval)
         /* Fallback to SGI extension. This is known to *not* support interval=0 */
         return (glXSwapIntervalSGI(interval) == 0);
     }
+#ifdef LLGL_LINUX_ENABLE_WAYLAND
+    }
+#endif
 
     return false;
 }
 
+#ifdef LLGL_LINUX_ENABLE_WAYLAND
```

Comment:
Can we split this into separate source files? One for X11 and one for Wayland, because this is a pretty large blend between X11 and Wayland code in the same source file. You don't have to split it into separate classes but do something similar to what I did in _LinuxGLSwapChain.cpp_ where I only moved the Linux specific code for the _GLSwapChain_ class into its own file.